vcr/util.py

```diff
@@ -92,6 +92,21 @@ def composed(incoming):
 def read_body(request):
     if hasattr(request.body, "read"):
         return request.body.read()
+    if hasattr(request.body, "__iter__") and not isinstance(
+        request.body,
+        (bytearray, bytes, dict, list, str),
```

Comment:
This very 5-types list exists in _two_ places in the code base now, so there is risk of the two going out of sync. Could this be (a) extracted to a non-public constant-like symbol somewhere or (b) commented about keeping in sync on both sides? (a) would likely be preferable. What do you think?

Reply:
added a new `is_iterator` function. WDYT?

Reply:
@sathieu the function idea and the function body I like, but the name is a bit misleading and I'm worried that this de-facto internal helper function is becoming part of the public API this way. How about `_is_nonsequence_iterator`? The naming idea is based on [`collections.abc.Sequence`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence) and it makes me realize as a side effect that we could make replace `isinstance(request.body, (bytearray, bytes, dict, list, str))` by `isinstance(request.body, Sequence)`. What do you think?

Reply:
Renamed to `_is_nonsequence_iterator`.

Unfortunately, `isinstance(request.body, Sequence)` is breaking a few tests...